tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -384,37 +395,69 @@ void transfer(
         writing to the consumer.
     */
 
+    volatile tt_l1_ptr CommandHeader* header = (CommandHeader*)command_ptr;
+    bool is_program = header->is_program_buffer;
+
     command_ptr += DeviceCommand::NUM_ENTRIES_IN_COMMAND_HEADER;
 
-    for (uint32_t i = 0; i < num_srcs; i++) {
+    for (uint32_t i = 0; i < num_buffer_transfers; i++) {
+        const uint32_t bank_base_address = command_ptr[0];
         const uint32_t num_pages = command_ptr[2];
         const uint32_t page_size = command_ptr[3];
+        const uint32_t src_buf_type = command_ptr[4];
+        const uint32_t dst_buf_type = command_ptr[5];
+        const uint32_t src_page_index = command_ptr[6];
+        uint32_t src_page_id = src_page_index;
+
+        bool skip_tx = forward_path & (BufferType)dst_buf_type == BufferType::SYSTEM_MEMORY & !is_program;
+        if (skip_tx) {
+            continue;
+        }
+
+        Buffer buffer;
+        bool read_local_buffer = forward_path & (BufferType)src_buf_type != BufferType::SYSTEM_MEMORY & is_program;
```

Comment:
A bit confused by this. Aka we want to read in dram pages for programs? but if that's case, not understanding src_buf_type = sysmem

Reply:
we check that src_buf_type != sysmem because there could be host data (rt args/cb configs) being tunnelled for